bm-persona/Dining/DiningDetailViewController.swift

```diff
@@ -0,0 +1,496 @@
+            openTag.backgroundColor = Color.openTag
+        } else {
+            openTag.text = "Closed"
+            openTag.backgroundColor = Color.closedTag
+        }
+        openTag.leftAnchor.constraint(equalTo: openTimeLabel.rightAnchor, constant: kViewMargin).isActive = true
+        openTag.centerYAnchor.constraint(equalTo: clockIcon.centerYAnchor).isActive = true
+        openTag.widthAnchor.constraint(equalToConstant: 50).isActive = true
+        
+        chairImage.centerYAnchor.constraint(equalTo: clockIcon.centerYAnchor).isActive = true
+        chairImage.leftAnchor.constraint(equalTo: openTag.rightAnchor, constant: kViewMargin).isActive = true
+        chairImage.widthAnchor.constraint(equalToConstant: 25).isActive = true
+        chairImage.heightAnchor.constraint(equalToConstant: 25).isActive = true
+        chairImage.contentMode = .scaleAspectFit
+
+        //TODO: use actual capacity
+        capBadge.text = "High"
+        capBadge.backgroundColor = Color.highCapacityTag
+        capBadge.leftAnchor.constraint(equalTo: chairImage.rightAnchor, constant: 5).isActive = true
+        capBadge.centerYAnchor.constraint(equalTo: clockIcon.centerYAnchor).isActive = true
+        capBadge.widthAnchor.constraint(equalToConstant: 50).isActive = true
+        capBadge.rightAnchor.constraint(lessThanOrEqualTo: faveButton.leftAnchor, constant: -1 * kViewMargin).isActive = true
+        
+        if diningHall!.isFavorited {
+            faveButton.setImage(UIImage(named: "Gold Star"), for: .normal)
+        } else {
+            faveButton.setImage(UIImage(named: "Grey Star"), for: .normal)
+        }
+        faveButton.centerYAnchor.constraint(equalTo: clockIcon.centerYAnchor).isActive = true
+        faveButton.rightAnchor.constraint(equalTo: card.layoutMarginsGuide.rightAnchor).isActive = true
+        faveButton.widthAnchor.constraint(equalToConstant: 25).isActive = true
+        faveButton.heightAnchor.constraint(equalToConstant: 25).isActive = true
+        
+        if diningHall!.image != nil {
+            diningImage.image = diningHall!.image
+        } else {
+            DispatchQueue.global().async {
+                guard let imageData = try? Data(contentsOf: self.diningHall!.imageURL!) else { return }
+                let image = UIImage(data: imageData)
+                DispatchQueue.main.async {
+                    self.diningHall!.image = image
+                    self.diningImage.image = image
+                }
+            }
+        }
+        diningImage.rightAnchor.constraint(equalTo: card.layoutMarginsGuide.rightAnchor).isActive = true
+        diningImage.topAnchor.constraint(equalTo: card.layoutMarginsGuide.topAnchor).isActive = true
+        diningImage.bottomAnchor.constraint(equalTo: faveButton.topAnchor, constant: -1 * kViewMargin).isActive = true
+        diningImage.widthAnchor.constraint(equalTo: card.layoutMarginsGuide.widthAnchor, multiplier: 0.45).isActive = true
```

Comment:
Would it be better to constrain the aspect ratio of the image? Unsure, so lmk.

Reply:
I think this is how you constrain the aspect ratio using anchors

Reply:
I mean the aspect ratio of the image width to the image height.

Reply:
oh yeah that's better